src/lsp/superbol_free_lib/command_check_syntax.ml

```diff
@@ -0,0 +1,51 @@
+(**************************************************************************)
+(*                                                                        *)
+(*  Copyright (c) 2021-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(*  All rights reserved.                                                  *)
+(*  This file is distributed under the terms of the                       *)
+(*  OCAMLPRO-NON-COMMERCIAL license.                                      *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** `parse` command *)
+
+open Ezcmd.V2
+open EZCMD.TYPES
+
+open Common_args
+
+let action { preproc_options; parser_options } files =
+  let parse input =
+    input |>
+    Cobol_preproc.preprocessor ~options:preproc_options |>
+    Cobol_parser.parse_simple ~options:parser_options
+  in
+  List.to_seq files |>
+  Seq.map begin fun filename ->
+    Pretty.out "@[Considering@ `%s'@]@." filename;
+    Cobol_preproc.Input.from ~filename ~f:parse
+  end
```

Comment:
With this, if multiple input file names are provided, we'd get a bunch of `Considering…`, and then all diagnostics gathered for every file.

Instead, directly using `List.iter` and then calling `Cobol_common.Diagnostics.sink_result` here would show diagnostics directly underneath the `Considering...` message. That would probably be better for a CLI.